src/core/net/ip6.cpp

```diff
@@ -426,6 +427,11 @@ ThreadError Ip6::HandleDatagram(Message &message, Netif *netif, int8_t interface
     {
         if (netif != NULL)
         {
+            if (otGetDeviceRole() == kDeviceRoleChild)
```

Comment:
I'm not sure adding this forwarding constraint is correct.  For example, a border router may be operating as a child and needs to forward IPv6 messages to/from the Thread network.